pyDeltaRCM/hook_tools.py

```diff
@@ -0,0 +1,169 @@
+        :obj:`~pyDeltaRCM.init_tools.init_tools.create_domain`.
+        """
+        pass
+
+    def hook_load_checkpoint(self):
+        """Hook :obj:`~pyDeltaRCM.init_tools.init_tools.load_checkpoint`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.init_tools.init_tools.load_checkpoint`.
+        """
+        pass
+
+    def hook_init_stratigraphy(self):
+        """Hook :obj:`~pyDeltaRCM.init_tools.init_tools.init_stratigraphy`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.init_tools.init_tools.init_stratigraphy`.
+        """
+        pass
+
+    def hook_run_one_timestep(self):
+        """Hook :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.run_one_timestep`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.run_one_timestep`.
+        """
+        pass
+
+    def hook_output_data(self):
+        """Hook :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.output_data`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.output_data`.
+        """
+        pass
+
+    def hook_apply_subsidence(self):
+        """Hook :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.apply_subsidence`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.apply_subsidence`.
+        """
+        pass
+
+    def hook_finalize_timestep(self):
+        """Hook :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.finalize_timestep`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.finalize_timestep`.
+        """
+        pass
+
+    def hook_output_checkpoint(self):
+        """Hook :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.output_checkpoint`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.iteration_tools.iteration_tools.output_checkpoint`.
+        """
+        pass
+
+    def hook_init_water_iteration(self):
+        """Hook :obj:`~pyDeltaRCM.water_tools.water_tools.init_water_iteration`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.water_tools.water_tools.init_water_iteration`.
+        """
+        pass
+
+    def hook_run_water_iteration(self):
+        """Hook :obj:`~pyDeltaRCM.water_tools.water_tools.run_water_iteration`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.water_tools.water_tools.run_water_iteration`.
+        """
+        pass
+
+    def hook_compute_free_surface(self):
+        """Hook :obj:`~pyDeltaRCM.water_tools.water_tools.compute_free_surface`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.water_tools.water_tools.compute_free_surface`.
+        """
+        pass
+
+    def hook_finalize_water_surface(self, iteration):
+        """Hook :obj:`~pyDeltaRCM.water_tools.water_tools.finalize_water_surface`.
+
+        Called immediately before
+        :obj:`~pyDeltaRCM.water_tools.water_tools.self`.
+        """
+        pass
```

Comment:
Consider changing this to be `hook_finalize_water_iteration` to match the function it is the hook for (`finalize_water_iteration`)?

Reply:
yeah, I'm not sure what I was thinking here hahah fixed.